File: 998-CheckCompletenessOfABinaryTree/998-CheckCompletenessOfABinaryTree.java
// Last updated: 8/5/2025, 2:56:14 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isCompleteTree(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while(queue.peek()!=null) {
            TreeNode node = queue.poll();
            queue.offer(node.left);
            queue.offer(node.right);
        }
        while(!queue.isEmpty()  && queue.peek()==null) {
            queue.poll();
        }
        return queue.isEmpty();
    }
}

File: 998-CheckCompletenessOfABinaryTree/998-CheckCompletenessOfABinaryTree.py
# Last updated: 8/5/2025, 2:56:16 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isCompleteTree(self, root: TreeNode) -> bool:
        bfs = [root]
        i = 0
        while bfs[i]:
            bfs.append(bfs[i].left)
            bfs.append(bfs[i].right)
            i += 1
        return not any(bfs[i:])